cashaddr.md

```diff
@@ -0,0 +1,117 @@
+# Address format for Bitcoin Cash
+
+Version 1.0, 2017-10-13
+
+## Abstract
+
+This document describes proposed address format to be used on Bitcoin Cash. It is a base32 encoded format using BCH codes as checksum and that can be used directly in links or QR codes.
+
+This format reuse the work done for Bech32[[1]](#bip173) and is similar is some aspects, but improves on others.
+
+## Specification
+
+The address is composed of
+1. A prefix indicating the network on which this address is valid.
+2. A separator, always `:`
+3. A base32 encoded payload indicating the destination of the address and containing a checksum.
+
+### Prefix
+
+The prefix indicates the network on which this addess is valid. It is set to `bitcoincash` for Bitcoin Cash main net, `xbctest` for bitcoin cash testnet and `xbcreg` for bitcoin cash regtest.
+
+The prefix is followed by the spearator `:`.
+
+When presented to users, the prefix may be omitted as it is part of the checksum computation. This ensure that addresses on different networks will remain incompatible, even in the absence of an explicit prefix.
+
+### Payload
+
+The payload is a base32 encoded stream of data.
+
+|     | 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 |
+| --: | - | - | - | - | - | - | - | - |
+|  +0 | q | p | z | r | y | 9 | x | 8 |
+|  +8 | g | f | 2 | t | v | d | w | 0 |
+| +16 | s | 3 | j | n | 5 | 4 | k | h |
+| +24 | c | e | 6 | m | u | a | 7 | l |
+
+The payload is composed of 3 elements:
+1. A version byte indicating the type of address.
+2. A hash.
+3. A 40 bits checksum.
+
+#### Version byte
+
+The version byte's most signficant bit is reserved and must be 0. The 4 next bits indicate the type of address and the 3 least significant bits indicate the size of the hash.
+
+| Size bits | Hash size in bits |
+| --------: | ----------------: |
+|         0 |               160 |
+|         1 |               192 |
+|         2 |               224 |
+|         3 |               256 |
+|         4 |               320 |
+|         5 |               384 |
+|         6 |               448 |
+|         7 |               512 |
+
+Encoding the size of the hash in the version field ensure that it is possible to check that the length of the address is correct.
+
+| Type bits |      Meaning      | Version byte value |
+| --------: | :---------------: | -----------------: |
+|         0 |       P2KH        |                  0 |
+|         1 |       P2SH        |                  8 |
+
+Further types will be added as news features are added.
+
+#### Hash
+
+The hash part really deserve not much explaination as its meaning is dependent on the version field. It is the hash that represents the data, namely a hash of the public key for P2KH and the hash of the reedemScript for P2SH.
```

Comment:
The part that readers will probably want to know is "how is this hash calculated".

Link to other spec would be good.

Some typo corrections:

deserve**s**

explanation